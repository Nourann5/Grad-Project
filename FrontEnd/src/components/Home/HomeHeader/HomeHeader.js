import React from 'react'
import { Col, Container, Row } from 'react-bootstrap'
import styles from './HomeHeader.module.css'

import {ReactComponent as JoinPortalIcon} from 'assets/icons/joinPortal.svg'
import {ReactComponent as MoreInfoIcon} from 'assets/icons/moreInfo.svg'
import imgSrc from 'assets/imgs/homePageHeaderImg.png'
import { Link } from 'react-router-dom'
function HomeHeader() {
  return (
    <header className={styles['header']}>
        <Container fluid='lg' className='h-100'>
            <Row className='h-100'>
                <Col lg='6' className='h-100'>
                    <div className={styles['header__content-cont']}>
                        <div className={styles['header__content-note-cont']}>
                            <p className={styles['header__content-note']}>
                                <span className={styles['header__content-note-title']}>New!</span>
                                Lorem ipsum dolor sit amet, consetetur sadipscing
                            </p>
                        </div>
                        
                        <div className={styles['header__content']}>
                            <h1 className={`${styles['header__content-title']}`}>
                                A NICE SECTION HEADING
                            </h1>
                            <p className={styles['header__content-desc']}>
                                Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut
                            </p>
                        </div>

                        <div className={styles['header__content-actions-cont']}>
                            <Link to='/register' className={`${styles["header__content-action-button"]}`}>
                                <JoinPortalIcon className={styles['header__content-actions-icon']}/>
                                join portal
                            </Link>
                            <button className={`${styles["header__content-action-button"]} ${styles["header__content-action--bg-dark"]}`}>
                                <MoreInfoIcon className={styles['header__content-actions-icon']}/>
                                More Info
                            </button>
                        </div>
                    </div>
                </Col>
                <Col lg='6'>
                    <div className={styles['header__img-cont']}>
                        <img src={imgSrc} alt='header img' className={styles['header__img']}/> 
                    </div>
                </Col>
            </Row>
        </Container>
    </header>
  )
}

export default HomeHeader