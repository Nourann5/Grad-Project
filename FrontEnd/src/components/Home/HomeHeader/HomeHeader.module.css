.header{
    /* background-color: red; */
    background-color: white;
    padding: 100px 0;
    /* min-height: 600px; */
}
.header__content-note{
    padding: 10px 40px;
    background-color: #F4F7FA;
    border-radius: 40px;
    margin-bottom: 30px;
    display: flex;
    align-items: center;
    font-size: 18px;
}
.header__content-note-title{
    background-color: #FFD101;
    border-radius: 27px;
    color: white;
    margin-right: 30px;
    font-size: 21px;
    font-family: 'PoppinsBoldItalic';
    padding: 10px 23px;
}
.header__content-title{
    margin-bottom: 25px;
    font-size: 35px;
    font-family: 'PoppinsBold';
}
.header__content-desc{
    font-size: 20px;
    font-family: 'PoppinsRegular';
    margin-bottom: 110px;
}

.header__content-action-button{
    background-color: var(--main-color);
    border-radius: 4px;
    padding: 18px 60px;
    color: white;
    margin-right: 20px;
    border:2px solid var(--main-color)!important;
}
.header__content-action-button:hover{
    color: white;
    opacity: .9;
}
.header__content-action--bg-dark{
    background-color: transparent;
    color: var(--main-color)!important;
    margin-right: 0;
}
.header__content-action--bg-dark:hover{
    color: var(--main-color);
    opacity: .9;
}
.header__content-actions-icon{
    margin-right: 10px;
    height: 24px;
}
.header__content-action-button svg path{
    fill:white
}
.header__content-action--bg-dark svg path{
    fill:var(--main-color)
}

.header__img-cont{
    height: 100%;
}
.header__img{
    height: 100%;
}

@media (max-width:992px) {
    .header{
        padding: 40px 0;
    }
    .header__content-desc{
        margin-bottom: 30px;
    }
    .header__img{
        display: none;
    }
    .header__content-actions-cont{
        display: flex;
        align-items: center;
        justify-content: space-around;
    }
    .header__content-action-button{
        padding: 8px 20px;
        margin-right: 10px;
    }
    .header__content-actions-icon{
        margin-right: 10px;
        height: 24px;
    } 
}