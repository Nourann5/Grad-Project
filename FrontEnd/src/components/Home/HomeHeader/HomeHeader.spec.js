import {render , screen} from '@testing-library/react'
import HomeHeader from './HomeHeader'
import { BrowserRouter, Route } from 'react-router-dom'

describe('Test the home header', () => { 
    it('heading and buttons exists',()=>{
        const tree =  render(
            <BrowserRouter>
                <HomeHeader/>
            </BrowserRouter>
        )
        const headerTitle = screen.getByRole('heading',{level:1})
        const headerMoreInfo = screen.getByRole('button')
        const headerPortalLink = screen.getByRole('link')
        expect(headerTitle).toBeInTheDocument()
        expect(headerPortalLink).toBeInTheDocument()
        expect(headerMoreInfo).toBeInTheDocument()
        expect(tree).toMatchSnapshot()
    })
 })