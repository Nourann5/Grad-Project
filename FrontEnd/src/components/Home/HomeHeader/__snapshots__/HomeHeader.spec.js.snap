// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the home header heading and buttons exists 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="h-100 container-lg"
        >
          <div
            class="h-100 row"
          >
            <div
              class="h-100 col-lg-6"
            >
              <div
                class="header__content-cont"
              >
                <div
                  class="header__content-note-cont"
                >
                  <p
                    class="header__content-note"
                  >
                    <span
                      class="header__content-note-title"
                    >
                      New!
                    </span>
                    Lorem ipsum dolor sit amet, consetetur sadipscing
                  </p>
                </div>
                <div
                  class="header__content"
                >
                  <h1
                    class="header__content-title"
                  >
                    A NICE SECTION HEADING
                  </h1>
                  <p
                    class="header__content-desc"
                  >
                    Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut
                  </p>
                </div>
                <div
                  class="header__content-actions-cont"
                >
                  <a
                    class="header__content-action-button"
                    href="/register"
                  >
                    <svg
                      class="header__content-actions-icon"
                    >
                      joinPortal.svg
                    </svg>
                    join portal
                  </a>
                  <button
                    class="header__content-action-button header__content-action--bg-dark"
                  >
                    <svg
                      class="header__content-actions-icon"
                    >
                      moreInfo.svg
                    </svg>
                    More Info
                  </button>
                </div>
              </div>
            </div>
            <div
              class="col-lg-6"
            >
              <div
                class="header__img-cont"
              >
                <img
                  alt="header img"
                  class="header__img"
                  src="homePageHeaderImg.png"
                />
              </div>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="h-100 container-lg"
      >
        <div
          class="h-100 row"
        >
          <div
            class="h-100 col-lg-6"
          >
            <div
              class="header__content-cont"
            >
              <div
                class="header__content-note-cont"
              >
                <p
                  class="header__content-note"
                >
                  <span
                    class="header__content-note-title"
                  >
                    New!
                  </span>
                  Lorem ipsum dolor sit amet, consetetur sadipscing
                </p>
              </div>
              <div
                class="header__content"
              >
                <h1
                  class="header__content-title"
                >
                  A NICE SECTION HEADING
                </h1>
                <p
                  class="header__content-desc"
                >
                  Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut
                </p>
              </div>
              <div
                class="header__content-actions-cont"
              >
                <a
                  class="header__content-action-button"
                  href="/register"
                >
                  <svg
                    class="header__content-actions-icon"
                  >
                    joinPortal.svg
                  </svg>
                  join portal
                </a>
                <button
                  class="header__content-action-button header__content-action--bg-dark"
                >
                  <svg
                    class="header__content-actions-icon"
                  >
                    moreInfo.svg
                  </svg>
                  More Info
                </button>
              </div>
            </div>
          </div>
          <div
            class="col-lg-6"
          >
            <div
              class="header__img-cont"
            >
              <img
                alt="header img"
                class="header__img"
                src="homePageHeaderImg.png"
              />
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
